### PR TITLE
Full support for extensions syntax.

### LINKED ISSUE #927: Wire doesn't accept multiple extension ranges declared in one `extensions` statement
The Protobuf spec allows multiple extension ranges to specified in a single `extensions` statement: [The spec specifies the following grammar:](https://developers.google.com/protocol-buffers/docs/reference/proto2-spec#extensions)

```
extensions = "extensions" ranges ";"
ranges = range { "," range }
range =  intLit [ "to" ( intLit | "max" ) ]
```

Here's an example protobuf message using this grammar:
```
message Msg {
    extensions 1, 3 to 5;
}
```

Google's protoc compiler accepts the above.
However, Wire only allows a single range; attempting to compile the above yields a syntax error. This discrepancy means a risk when using both Wire and Google's protoc in the same code base, or when trying to move from one to the other.

I'm not familiar with the Wire codebase, but it looks like the parsing code in [ProtoParser.readExtensions](https://github.com/square/wire/blob/master/wire-schema/src/main/java/com/squareup/wire/schema/internal/parser/ProtoParser.java#L471) is what doesn't allow multiple ranges. I'm not sure what other pieces of code may need to change to accommodate multiple ranges.

### PR DESCRIPTION
**Pull request by oldergod**
fixes #927

### CODE REVIEW

File: wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt [hunk 1]
```diff
@@ -22,23 +22,38 @@ import kotlin.jvm.JvmStatic
 class Extensions private constructor(
   val location: Location,
   val documentation: String,
-  val start: Int,
-  val end: Int
+  val values: List<Any>
```
Comment (swankjesse) [inline wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt]: document what the potential values are?

Comment (swankjesse) [reply wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt]: I guess that’s implied on line 31

File: wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt [hunk 2]
```diff
@@ -22,23 +22,38 @@ import kotlin.jvm.JvmStatic
...
 ) {
   fun validate(linker: Linker) {
-    if (!start.isValidTag() || !end.isValidTag()) {
-      linker.withContext(this)
-          .addError("tags are out of range: $start to $end")
+    val linker = linker.withContext(this)
+    val outOfRangeTags = mutableListOf<String>()
+    values.forEach { value ->
```
Comment (swankjesse) [inline wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt]: why not use a for loop?

File: wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt [hunk 3]
```diff
@@ -22,23 +22,38 @@ import kotlin.jvm.JvmStatic
...
+      when (value) {
+        is Int -> {
+          if (!value.isValidTag()) {
+            outOfRangeTags.add("$value")
+          }
+        }
+        is IntRange -> {
+          if (!value.first.isValidTag() || !value.last.isValidTag()) {
```
Comment (swankjesse) [inline wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt]: check they’re non-decreasing?

File: wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt (hunk 1 above)
Comment (swankjesse) [reply wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt]: ahh, found it in the element

Comment (oldergod) [reply wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt]: Yeah, it wrote it the same way `ReservedElement` was documented.

File: wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt (hunk 2 above)
Comment (oldergod) [reply wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt]: no reason really

File: wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt (hunk 3 above)
Comment (oldergod) [reply wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extensions.kt]: Why not

**Pull request merged**